src/interface.jl

```diff
@@ -159,9 +162,21 @@ function (f::PrimeDerivativeBack)(x)
     z = ∂⃖¹(lower_pd(f), x)
     y = getfield(z, 1)
     f☆ = getfield(z, 2)
-    return getfield(f☆(dx(y)), 2)
+    return unthunk(getfield(f☆(dx(y)), 2))
+end
+
+
+struct JacobianBack{N, T} <: AbstractPrimeDerivative{N, T}
+    f::T
+end
+
+function (f::JacobianBack)(x)
+    y, f☆ = ∂⃖(getfield(f, :f), x)
+    return transpose(reduce(hcat, f☆(collect(row))[2] for row in dx_jacobian(y)))
```

Comment:
I think some `vec` may be missing:
```
julia> Diffractor.jacobian(x -> x[1:2])(rand(3,3))
6×3 transpose(::Matrix{Float64}) with eltype Float64:
 1.0  0.0  0.0
 0.0  0.0  0.0
 0.0  0.0  0.0
 0.0  1.0  0.0
 0.0  0.0  0.0
 0.0  0.0  0.0

julia> Zygote.jacobian(x -> x[1:2], rand(3,3))[1]
2×9 Matrix{Float64}:
 1.0  0.0  0.0  0.0  0.0  0.0  0.0  0.0  0.0
 0.0  1.0  0.0  0.0  0.0  0.0  0.0  0.0  0.0
```
And maybe a `tuple`?
```
julia> Diffractor.gradient(atan)(1/2)
(0.8,)

julia> Diffractor.gradient(atan)(1, 2)
(0.4, -0.2)

julia> Diffractor.jacobian(x -> x[1:2])([3,4,5])  # shouldn't this also be a tuple?
2×3 transpose(::Matrix{Int64}) with eltype Int64:
 1  0  0
 0  1  0

julia> Diffractor.jacobian(*)(rand(2,2), rand(2))  # to allow this...
ERROR: MethodError: no method matching (::Diffractor.JacobianBack{1, typeof(*)})(::Matrix{Float64}, ::Vector{Float64})
```

Reply:
Or maybe jacobian deserves to be another PR. What should these do?
```
julia> Diffractor.PrimeDerivativeBack(x -> sum(sum(x.a) .+ x.b))((a=[1,2], b=[3,4], c=[5,6]))  # maybe this should unthunk within Tangent? 
Tangent{NamedTuple{(:a, :b, :c), Tuple{Vector{Int64}, Vector{Int64}, Vector{Int64}}}}(b = [1.0, 1.0], a = InplaceableThunk(ChainRules.var"#..., Thunk(ChainRules.var"#...)))

julia> Diffractor.gradient(x -> sum(sum(x.a) .+ x.b), (a=[1,2], b=[3,4], c=[5,6]))[1]  # ditto?
Tangent{NamedTuple{(:a, :b, :c), Tuple{Vector{Int64}, Vector{Int64}, Vector{Int64}}}}(b = [1.0, 1.0], a = InplaceableThunk(ChainRules.var"#..., Thunk(ChainRules.var"#...)))

julia> Diffractor.jacobian(x -> sum(x.a) .+ x.b)((a=[1,2], b=[3,4], c=[5,6]))  # what structure should this return?
2×1 transpose(::Matrix{Tangent{NamedTuple{(:a, :b, :c), Tuple{Vector{Int64}, Vector{Int64}, Vector{Int64}}}, NamedTuple{(:b, :a), Tuple{Vector{Int64}, InplaceableThunk{Thunk{ChainRules.var"#1573#1576"{Int64, Colon, Vector{Int64}, ProjectTo{AbstractArray, NamedTuple{(:element, :axes), Tuple{ProjectTo{Float64, NamedTuple{(), Tuple{}}}, Tuple{Base.OneTo{Int64}}}}}}}, ChainRules.var"#1572#1575"{Int64, Colon}}}}}}) with eltype Union{}:
Error showing value of type Transpose{Union{}, Matrix{Tangent{NamedTuple{(:a, :b, :c), Tuple{Vector{Int64}, Vector{Int64}, Vector{Int64}}}, NamedTuple{(:b, :a), Tuple{Vector{Int64}, InplaceableThunk{Thunk{ChainRules.var"#1573#1576"{Int64, Colon, Vector{Int64}, ProjectTo{AbstractArray, NamedTuple{(:element, :axes), Tuple{ProjectTo{Float64, NamedTuple{(), Tuple{}}}, Tuple{Base.OneTo{Int64}}}}}}}, ChainRules.var"#1572#1575"{Int64, Colon}}}}}}}:
ERROR: MethodError: no method matching transpose(::Tangent{NamedTuple{(:a, :b, :c), Tuple{Vector{Int64}, Vector{Int64}, Vector{Int64}}}, NamedTuple{(:b, :a), Tuple{Vector{Int64}, InplaceableThunk{Thunk{ChainRules.var"#1573#1576"{Int64, Colon, Vector{Int64}, ProjectTo{AbstractArray, NamedTuple{(:element, :axes), Tuple{ProjectTo{Float64, NamedTuple{(), Tuple{}}}, Tuple{Base.OneTo{Int64}}}}}}}, ChainRules.var"#1572#1575"{Int64, Colon}}}}})
```

Reply:
honestly I meant to make it a separate PR and forgot to check out a new PR.

Reply:
Can this branch `unthunk` apply to `gradient` too, and perhaps within `Tangent`?